relayer/src/tx/connection.rs

```diff
@@ -67,7 +76,140 @@ pub fn conn_init(opts: &ConnectionOpenInitOptions) -> Result<Vec<u8>, Error> {
         signer,
     };
 
-    let proto_msgs: Vec<Any> = vec![new_msg.to_any::<RawMsgConnectionOpenInit>()];
+    Ok(vec![new_msg.to_any::<RawMsgConnectionOpenInit>()])
+}
+
+pub fn build_conn_init_and_send(opts: &ConnectionOpenInitOptions) -> Result<String, Error> {
+    // Get the source and destination chains.
+    let src_chain = &CosmosSDKChain::from_config(opts.clone().src_chain_config)?;
+    let dest_chain = &mut CosmosSDKChain::from_config(opts.clone().dest_chain_config)?;
+
+    let new_msgs = build_conn_init(dest_chain, src_chain, opts)?;
+    let (key, _) = dest_chain.key_and_signer(&opts.signer_seed)?;
+
+    Ok(dest_chain.send(new_msgs, key, "".to_string(), 0)?)
+}
+
+#[derive(Clone, Debug)]
+pub struct ConnectionOpenTryOptions {
+    pub dest_chain_config: ChainConfig,
+    pub src_chain_config: ChainConfig,
+    pub dest_client_id: ClientId,
+    pub src_client_id: ClientId,
+    pub dest_connection_id: ConnectionId,
+    pub src_connection_id: ConnectionId,
+    pub signer_seed: String,
+}
+
+fn check_connection_state_for_try(
+    connection_id: ConnectionId,
+    existing_connection: ConnectionEnd,
+    expected_connection: ConnectionEnd,
+) -> Result<(), Error> {
+    if existing_connection != expected_connection {
+        Err(Kind::ConnOpenTry(
+            connection_id,
+            "connection already exist in an incompatible state".into(),
+        )
+        .into())
+    } else {
+        Ok(())
+    }
+}
+
+/// Attempts to send a MsgConnOpenTry to the dest_chain.
+pub fn build_conn_try(
+    dest_chain: &mut CosmosSDKChain,
+    src_chain: &CosmosSDKChain,
+    opts: &ConnectionOpenTryOptions,
+) -> Result<Vec<Any>, Error> {
+    // If there is a connection present on the destination chain it should look like this
+    let counterparty = Counterparty::new(
+        opts.src_client_id.clone(),
+        Some(opts.src_connection_id.clone()),
+        src_chain.query_commitment_prefix()?,
+    );
+    let dest_expected_connection = ConnectionEnd::new(
+        State::Init,
+        opts.dest_client_id.clone(),
+        counterparty.clone(),
+        src_chain.query_compatible_versions()?,
+    )
+    .unwrap();
+
+    // Check that if a connection exists on destination it is consistent with the try options
+    if let Ok(dest_connection) =
```

Comment:
Just a remark: This verification if very similar to the one in the ICS03/try handler:

https://github.com/informalsystems/ibc-rs/blob/30422b1f793fcb0b0ca2a95a9d0e11b45a38f548/modules/src/ics03_connection/handler/conn_open_try.rs#L31-L48

So there may be room to factor these commonalities in a single method and reuse that method in both the handler and the relayer. We should do that later, however (once we see a bit more tx raws); this issue is dedicated to doing that: https://github.com/informalsystems/ibc-rs/issues/92

Reply:
Yes, agreed. I also noticed that and looked a bit on how the code should change...again pretty big refactoring so let's do it in a separate PR.